### PR TITLE
Use Debian 12 image for building

### PR DESCRIPTION
**Pull request by pstranak-sw**
Replaces https://github.com/solarwinds/swi-k8s-opentelemetry-collector/pull/339

### CODE REVIEW

File: build/docker/structure-test.yaml [hunk 1]
```diff
@@ -14,8 +14,7 @@ commandTests:
   - name: "journalctl is working in the image"
     command: "journalctl"
     expectedOutput: ["-- No entries --\n"]
-    excludedError: ["No journal files were found.\n"]
   - name: "swi-otelcol is working in the image"
     command: "/swi-otelcol"
     args: ["-v"]
-    expectedOutput: ["swi-k8s-opentelemetry-collector version 0.8.4"]
\ No newline at end of file
+    expectedOutput: ["swi-k8s-opentelemetry-collector version 0.8.5"]
```
Comment (pstranak-sw) [inline build/docker/structure-test.yaml:20]: This was forgotten in one of the previous commits and is not directly related to the goal of this PR.

File: build/docker/structure-test.yaml [hunk 2]
```diff
@@ -14,8 +14,7 @@ commandTests:
   - name: "journalctl is working in the image"
     command: "journalctl"
     expectedOutput: ["-- No entries --\n"]
-    excludedError: ["No journal files were found.\n"]
```
Comment (pstranak-sw) [inline build/docker/structure-test.yaml:17]: Removing the test. The `stderr` of the `journalctl` contained the `No journal files were found.` message even before this PR.
For some reason, the test did not detect it before. Now, with the new `journalctl` binary, the message is rendered differently in the terminal and the test started detecting it.

**Review by gantrior — APPROVED** [review]
